### DISCUSSION TITLE
Channel Debugger not working

### DISCUSSION THREAD

**Discussion by Guomin42** [nextgenhealthcare/connect#5948] [Q&A] [answered]
Hi everyone, I was trying out the channel debugger on a channel. After I clicked the OK button on the Debug Channel Popup dialogue box, the mirth server throwed an exception on the server log as following:
![image](https://github.com/nextgenhealthcare/connect/assets/13042868/10a2dce6-3639-4f15-996f-99c2e048b2af)
![image](https://github.com/nextgenhealthcare/connect/assets/13042868/bde8fd2c-420d-4c12-8a0d-877eb5b7450b)

Linux debian-kernel
Zulu OpenJDK 17
Mirth 4.4 installed using a dockerfile with awt headless config being changed on mcserver.vmoptions to false.

**Comment by pacmano1** [comment]
The debugger will open on the server running the engine, it won't open remotely on your desktop.  So that won't work with Docker.

**Comment by pacmano1** [comment, marked as answer]
See https://github.com/nextgenhealthcare/connect/wiki/4.0.0---What's-New#javascript-debugger

"It's important to understand that the Debugger and its user interface runs on the server side".

The user guide is not really clear on that.
